pkg/webhook/webhook.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"os"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/webhook"
+	"github.com/openshift/generic-admission-server/pkg/apiserver"
+	"github.com/openshift/generic-admission-server/pkg/cmd/server"
+	"github.com/spf13/cobra"
+)
+
+func NewWebhookCommand(stopCh <-chan struct{}) *cobra.Command {
+	admissionHooks := []apiserver.AdmissionHook{
+		&webhook.FederatedTypeConfigValidationHook{},
+		&webhook.FederatedClusterValidationHook{},
+	}
+
+	// done to avoid cannot use admissionHooks (type []AdmissionHook) as type []apiserver.AdmissionHook in argument to "github.com/openshift/kubernetes-namespace-reservation/pkg/genericadmissionserver/cmd/server".NewCommandStartAdmissionServer
```

Comment:
Do not understand this.
The variables at line 29 and that at line 35 are of same type. Also `NewCommandStartAdmissionServer` seems to take the same type. Am I missing something?

Reply:
This seems entirely unnecessary to me.